docs/migration/cosmoshub-4-v10-upgrade.md

```diff
@@ -0,0 +1,295 @@
+
+#### Preparation
+
+Install the latest version of Cosmovisor (`1.3.0`):
+
+```shell
+go install github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor@latest
+```
+
+Create a cosmovisor folder:
+
+create a cosmovisor folder inside gaia home and move gaiad v9.1.1 into `$GAIA_HOME/cosmovisor/genesis/bin`
+
+```shell
+mkdir -p $GAIA_HOME/cosmovisor/genesis/bin
+cp $(which gaiad) $GAIA_HOME/cosmovisor/genesis/bin
+```
+
+```shell
+.
+├── current -> genesis or upgrades/<name>
+└── genesis
+     └── bin
+        └── gaiad  #v9.1.1
+```
+
+Export the environmental variables:
+
+```shell
+export DAEMON_NAME=gaiad
+# please change to your own gaia home dir
+export DAEMON_HOME=$GAIA_HOME
+export DAEMON_RESTART_AFTER_UPGRADE=true
+export DAEMON_ALLOW_DOWNLOAD_BINARIES=true
+```
+
+Start the node:
+
+```shell
+cosmovisor run start --x-crisis-skip-assert-invariants --home $DAEMON_HOME
+```
+
+Skipping the invariant checks can help decrease the upgrade time significantly.
+
+#### Expected result
+
+When the upgrade block height is reached, you can find the following information in the log:
+
+```shell
+ERR UPGRADE "v10" NEEDED at height: 15816200: upgrade to v10 and applying upgrade "v10" at height:15816200
+```
+
+Then the Cosmovisor will create `$GAIA_HOME/cosmovisor/upgrades/v10/bin` and download the Gaia v10.0.0 binary to this folder according to links in the `--info` field of the upgrade proposal.
+This may take 7 minutes to a few hours, afterwards, the chain will continue to produce blocks once validators with a total sum voting power > 2/3 complete their nodes upgrades.
+
+_Please Note:_
+
+- In general, auto-download comes with the risk that the verification of correct download is done automatically. If users want to have the highest guarantee users should confirm the check-sum manually. We hope more node operators will use the auto-download for this release but please be aware this is a risk and users should take at your own discretion.
+- Users should use run node on v9.1.1 if they use the cosmovisor v1.3.0 with auto-download enabled for upgrade process.
+
+## Upgrade duration
+
+The upgrade may take a few minutes to several hours to complete because cosmoshub-4 participants operate globally with differing operating hours and it may take some time for operators to upgrade their binaries and connect to the network.
+
+## Rollback plan
+
+During the network upgrade, core Cosmos teams will be keeping an ever vigilant eye and communicating with operators on the status of their upgrades. During this time, the core teams will listen to operator needs to determine if the upgrade is experiencing unintended challenges. In the event of unexpected challenges, the core teams, after conferring with operators and attaining social consensus, may choose to declare that the upgrade will be skipped.
+
+Steps to skip this upgrade proposal are simply to resume the cosmoshub-4 network with the (downgraded) v8.0.1 binary using the following command:
```

Comment:
Should this say "(downgraded) v9.1.1 binary"?

Reply:
Yep, your right.